4 tests fail for =dask-1.1.3

I am packaging this new version of your software for Gentoo Linux, and I am running into the following errors in the test suite:


```
=============================================== FAILURES ================================================
_____________________________________ test_reductions_frame_dtypes ______________________________________

    def test_reductions_frame_dtypes():
        df = pd.DataFrame({'int': [1, 2, 3, 4, 5, 6, 7, 8],
                           'float': [1., 2., 3., 4., np.nan, 6., 7., 8.],
                           'dt': [pd.NaT] + [datetime(2011, i, 1) for i in range(1, 8)],
                           'str': list('abcdefgh')})
    
        if HAS_INT_NA:
            df['intna'] = pd.array([1, 2, 3, 4, None, 6, 7, 8], dtype=pd.Int64Dtype())
    
        ddf = dd.from_pandas(df, 3)
        assert_eq(df.sum(), ddf.sum())
        assert_eq(df.prod(), ddf.prod())
        assert_eq(df.min(), ddf.min())
        assert_eq(df.max(), ddf.max())
        assert_eq(df.count(), ddf.count())
        assert_eq(df.std(), ddf.std())
        assert_eq(df.var(), ddf.var())
        assert_eq(df.sem(), ddf.sem())
        assert_eq(df.std(ddof=0), ddf.std(ddof=0))
        assert_eq(df.var(ddof=0), ddf.var(ddof=0))
        assert_eq(df.sem(ddof=0), ddf.sem(ddof=0))
    
        result = ddf.mean()
        expected = df.mean()
    
        if PANDAS_GT_0240:
            # https://github.com/pandas-dev/pandas/issues/24752
            # It's not clear whether this will be included by
            # default.
>           expected = expected.drop('dt').astype(float)

dask/dataframe/tests/test_arithmetics_reduction.py:1015: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/lib64/python2.7/site-packages/pandas/core/series.py:3897: in drop
    inplace=inplace, errors=errors)
/usr/lib64/python2.7/site-packages/pandas/core/generic.py:3782: in drop
    obj = obj._drop_axis(labels, axis, level=level, errors=errors)
/usr/lib64/python2.7/site-packages/pandas/core/generic.py:3814: in _drop_axis
    new_axis = axis.drop(labels, errors=errors)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Index([u'float', u'int', u'intna'], dtype='object'), labels = array(['dt'], dtype=object)
errors = 'raise'

    def drop(self, labels, errors='raise'):
        """
        Make new Index with passed list of labels deleted.
    
        Parameters
        ----------
        labels : array-like
        errors : {'ignore', 'raise'}, default 'raise'
            If 'ignore', suppress error and existing labels are dropped.
    
        Returns
        -------
        dropped : Index
    
        Raises
        ------
        KeyError
            If not all of the labels are found in the selected axis
        """
        arr_dtype = 'object' if self.dtype == 'object' else None
        labels = com.index_labels_to_array(labels, dtype=arr_dtype)
        indexer = self.get_indexer(labels)
        mask = indexer == -1
        if mask.any():
            if errors != 'ignore':
                raise KeyError(
>                   '{} not found in axis'.format(labels[mask]))
E               KeyError: "['dt'] not found in axis"

/usr/lib64/python2.7/site-packages/pandas/core/indexes/base.py:4981: KeyError
________________________ test_repartition_npartitions[<lambda>0-float-5-1-True] _________________________

use_index = True, n = 1, k = 5, dtype = <type 'float'>, transform = <function <lambda> at 0x7f59d00999b0>

    @pytest.mark.parametrize('use_index', [True, False])
    @pytest.mark.parametrize('n', [1, 2, 4, 5])
    @pytest.mark.parametrize('k', [1, 2, 4, 5])
    @pytest.mark.parametrize('dtype', [int, float, 'M8[ns]'])
    @pytest.mark.parametrize('transform', [lambda df: df, lambda df: df.x])
    def test_repartition_npartitions(use_index, n, k, dtype, transform):
        df = pd.DataFrame({'x': [1, 2, 3, 4, 5, 6] * 10,
                           'y': list('abdabd') * 10},
                          index=pd.Series([10, 20, 30, 40, 50, 60] * 10, dtype=dtype))
        df = transform(df)
        a = dd.from_pandas(df, npartitions=n, sort=use_index)
        b = a.repartition(npartitions=k)
        assert_eq(a, b)
        assert b.npartitions == k
        parts = dask.get(b.dask, b.__dask_keys__())
>       assert all(map(len, parts))
E       assert False
E        +  where False = all([10, 10, 20, 0, 20])
E        +    where [10, 10, 20, 0, 20] = map(len, (      x  y\n10.0  1  a\n10.0  1  a\n10.0  1  a\n10.0  1  a\n10.0  1  a\n10.0  1  a\n10.0  1  a\n10.0  1  a\n10.0  1  a\n10.0  1...b\n50.0  5  b\n50.0  5  b\n50.0  5  b\n5... d\n60.0  6  d\n60.0  6  d\n60.0  6  d\n60.0  6  d\n60.0  6  d\n60.0  6  d\n60.0  6  d))

dask/dataframe/tests/test_dataframe.py:1357: AssertionError
________________________ test_repartition_npartitions[<lambda>1-float-5-1-True] _________________________

use_index = True, n = 1, k = 5, dtype = <type 'float'>, transform = <function <lambda> at 0x7f59d0099a28>

    @pytest.mark.parametrize('use_index', [True, False])
    @pytest.mark.parametrize('n', [1, 2, 4, 5])
    @pytest.mark.parametrize('k', [1, 2, 4, 5])
    @pytest.mark.parametrize('dtype', [int, float, 'M8[ns]'])
    @pytest.mark.parametrize('transform', [lambda df: df, lambda df: df.x])
    def test_repartition_npartitions(use_index, n, k, dtype, transform):
        df = pd.DataFrame({'x': [1, 2, 3, 4, 5, 6] * 10,
                           'y': list('abdabd') * 10},
                          index=pd.Series([10, 20, 30, 40, 50, 60] * 10, dtype=dtype))
        df = transform(df)
        a = dd.from_pandas(df, npartitions=n, sort=use_index)
        b = a.repartition(npartitions=k)
        assert_eq(a, b)
        assert b.npartitions == k
        parts = dask.get(b.dask, b.__dask_keys__())
>       assert all(map(len, parts))
E       assert False
E        +  where False = all([10, 10, 20, 0, 20])
E        +    where [10, 10, 20, 0, 20] = map(len, (10.0    1\n10.0    1\n10.0    1\n10.0    1\n10.0    1\n10.0    1\n10.0    1\n10.0    1\n10.0    1\n10.0    1\nName: x, dtype: i....0    5\n50.0    5\n50.0    5\n50.0    ....0    6\n60.0    6\n60.0    6\n60.0    6\n60.0    6\n60.0    6\nName: x, dtype: int64))

dask/dataframe/tests/test_dataframe.py:1357: AssertionError
_____________________________________ test_register_extension_type ______________________________________

    def test_register_extension_type():
        arr = DecimalArray._from_sequence([Decimal('1.0')] * 10)
        ser = pd.Series(arr)
        dser = dd.from_pandas(ser, 2)
>       assert_eq(ser, dser)

dask/dataframe/tests/test_extensions.py:27: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
dask/dataframe/utils.py:686: in assert_eq
    b = _check_dask(b, check_names=check_names, check_dtypes=check_dtypes)
dask/dataframe/utils.py:615: in _check_dask
    result = dsk.compute(scheduler='sync')
dask/base.py:156: in compute
    (result,) = compute(self, traverse=False, **kwargs)
dask/base.py:399: in compute
    return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
dask/dataframe/core.py:78: in finalize
    return _concat(results)
dask/dataframe/core.py:74: in _concat
    return args[0] if not args2 else methods.concat(args2, uniform=True)
dask/dataframe/methods.py:271: in concat
    filter_warning=filter_warning)
dask/dataframe/methods.py:381: in concat_pandas
    out = pd.concat(dfs2, join=join, **concat_kwargs)
/usr/lib64/python2.7/site-packages/pandas/core/reshape/concat.py:229: in concat
    return op.get_result()
/usr/lib64/python2.7/site-packages/pandas/core/reshape/concat.py:393: in get_result
    self.new_axes)
/usr/lib64/python2.7/site-packages/pandas/core/internals/managers.py:1616: in concat
    if len({b.dtype for b in blocks}) == 1:
/usr/lib64/python2.7/site-packages/pandas/core/internals/managers.py:1616: in <setcomp>
    if len({b.dtype for b in blocks}) == 1:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = DecimalDtype(context=Context(prec=28, rounding=ROUND_HALF_EVEN, Emin=-99999999...999, capitals=1, flags=[], traps=[DivisionByZero, InvalidOperation, Overflow]))

    def __hash__(self):
>       return hash(tuple(getattr(self, attr) for attr in self._metadata))
E       TypeError: unhashable type: 'Context'
```

Full [build.log](https://github.com/dask/dask/files/2935562/build.log).

Any idea what I can do about this (or if it's safe to just xfail the tests)?

--- a/.travis.yml
+++ b/.travis.yml
@@ -45,7 +45,7 @@ jobs:
     - env: &py37_env
       - PYTHON=3.7
       - NUMPY=1.15.0
-      - PANDAS=0.23.4
+      - PANDAS>=0.24.1
       - *test_and_lint
       - *no_coverage
       - *no_optimize

--- a/dask/dataframe/tests/test_arithmetics_reduction.py
+++ b/dask/dataframe/tests/test_arithmetics_reduction.py
@@ -8,7 +8,6 @@ import pandas as pd
 import dask.dataframe as dd
 from dask.dataframe.utils import (
     assert_eq, assert_dask_graph, make_meta, HAS_INT_NA,
-    PANDAS_GT_0240
 )
 
 
@@ -1007,13 +1006,6 @@ def test_reductions_frame_dtypes():
 
     result = ddf.mean()
     expected = df.mean()
-
-    if PANDAS_GT_0240:
-        # https://github.com/pandas-dev/pandas/issues/24752
-        # It's not clear whether this will be included by
-        # default.
-        expected = expected.drop('dt').astype(float)
-
     assert_eq(expected, result)
 
     assert_eq(df._get_numeric_data(), ddf._get_numeric_data())

--- a/dask/dataframe/tests/test_extensions.py
+++ b/dask/dataframe/tests/test_extensions.py
@@ -3,6 +3,7 @@ import pytest
 
 import dask.dataframe as dd
 from dask.dataframe.utils import assert_eq, PANDAS_VERSION
+from dask.compatibility import PY2
 
 pd = pytest.importorskip("pandas", minversion="0.23.4")
 
@@ -20,6 +21,7 @@ def _(dtype):
                                        **kwargs)
 
 
+@pytest.mark.skipif(PY2, reason="unhashable Context")
 def test_register_extension_type():
     arr = DecimalArray._from_sequence([Decimal('1.0')] * 10)
     ser = pd.Series(arr)

